sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -196,6 +196,90 @@
  *    <li>Since version: 3.4.0</li>
  *   </ul>
  *  </li>
+ *  <li>Name: <code>DATE_ADD</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>DATE_ADD(start_date, num_days)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>DATE_DIFF</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>DATE_DIFF(end_date, start_date)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>TRUNC</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>TRUNC(date, format)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>SECOND</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>SECOND(timestamp, timezone)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>MINUTE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>MINUTE(timestamp, timezone)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>HOUR</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>HOUR(timestamp, timezone)</code></li>
+ *    <li>Since version: 3.4.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>MONTH</code>
```

Comment:
I'm wondering if the `EXTRACT` function is more common that these field extraction functions?

Reply:
Yes. In some databases these field extractions are deprecated and EXTRACT is used instead, such as H2.
http://h2database.com/html/functions.html#month

Reply:
Can we generate EXTRACT function instead? I'm more comfortable exposing some Spark-specific fields in the EXTRACT function, instead of exposing spark-specific functions.

Reply:
But in different databases, the performance of EXTRACT function and field extraction functions is different. For example, pg, mysql, snowflake and the like are used together. And something like clickhouse only uses the field extraction functions.

Reply:
I'm surprised if the EXTRACT function has different performance than field extraction functions. They are semantically doing the same thing, right?

The principle here is to not define push-able functions that are semantically equal to each other. Keep in mind that this is API, and it's better to have less APIs but the same abilities. I think adding a single `EXTRACT` function to the pushdown API is better than adding a bunch of the field extraction functions.

clickhouse data source can translate EXTRACT function to field extraction functions when generating SQL string